docs/network-policy.md

```diff
@@ -0,0 +1,149 @@
+ensure the behavior they expect.
+
+**ingress**: Each ClusterNetworkPolicy may consist of zero or more ordered
+set of ingress rules. Each rule, depending on the `action` field of the rule,
+allows or drops traffic which matches both the `from` and `ports` sections.
+The example policy contains a single rule, which allows matched traffic on a
+single port, from one of two sources: the first specified by a `podSelector`
+and the second specified by a combination of a `podSelector` and a
+`namespaceSelector`.
+**Note**: The order in which the ingress rules are set matter, i.e. rules will be
+evaluated in the order in which they are written.
+
+**egress**: Each ClusterNetworkPolicy may consist of zero or more ordered set of
+egress rules. Each rule, depending on the `action` field of the rule, allows
+or drops traffic which matches both the `to` and `ports` sections. The example
+policy contains a single rule, which drops matched traffic on a single port,
+to the 10.0.10.0/24 subnet specified by the `ipBlock` field.
+**Note**: The order in which the egress rules are set matter, i.e. rules will be
+evaluated in the order in which they are written.
+
+## Rule evaluation based on priorities
+
+Rules belonging to Cluster NetworkPolicy CRDs are associated with various
+priorities, such as the `priority` at the CNP level and the priority at rule
+level. Overall, Cluster Policy with highest precedence (lowest priority number
+value) is evaluated first. Within this policy, rules are evaluated in the order
+in which they are set. For example, consider the following:
+
+- CNP1{priority: 10, ingressRules: [ir1.1, ir1.2], egressRules: [er1.1, er1.2]}
+- CNP2{priority: 15, ingressRules: [ir2.1, ir2.2], egressRules: [er2.1, er2.2]}
+
+This translates to the following order:
+- Ingress rules: ir1.1 -> ir1.2 -> ir2.1 -> ir2.2
+- Egress rules: er1.1 -> er1.2 -> er2.1 -> er2.2
+
+Once a rule is matched, it is executed based on the action set. If none of the
+CNP rules match, the packet is then evaluated for rules created for K8s NP.
+Hence, CNP take precedence over K8s NP.
+
+## Behavior of `to` and `from` selectors
+
+There are four kinds of selectors that can be specified in an ingress `from`
+section or egress `to` section:
+
+**podSelector**: This selects particular Pods from all Namespaces as `sources`,
+if set in `ingress` section, or as `destinations`, if set in `egress` section.
+
+**namespaceSelector**: This selects particular Namespaces for which all Pods are
+grouped as `ingress` `sources` or `egress` `destinations`.
+
+**podSelector** and **namespaceSelector**:  A single to/from entry that specifies
+both namespaceSelector and podSelector selects particular Pods within
+particular Namespaces. 
+
+**ipBlock**: This selects particular IP CIDR ranges to allow as `ingress` `sources`
+or `egress` `destinations`. These should be cluster-external IPs, since Pod IPs are
+ephemeral and unpredictable.
```

Comment:
why are we highlighting `sources` and `destinations` here? I don't think they are field names in the CRD?

Reply:
that's a mistake.. updated.. also added a note regarding alpha status